src/test/java/io/wgj/ctci/StackTest.java

```diff
@@ -2,49 +2,121 @@
 
 import static org.junit.Assert.*;
 
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 
-import java.util.ArrayList;
-import java.util.Collections;
 
 public class StackTest {
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
     @Test
     public void testStackOneInt() {
-        Stack s = new Stack();
+        Stack<Integer> s = new Stack<>();
         s.push(1);
         int r = s.pop();
         assertEquals(1, r);
     }
 
+    @Test
+    public void testStackOneString() {
+        Stack<String> s = new Stack<>();
+        s.push("one");
+        String r = s.pop();
+        assertEquals("one", r);
+    }
+
+    @Test
+    public void testPushPopsIntermixed() {
+        Stack<Integer> s = new Stack<>();
+        s.push(0);
+        assertEquals(0, (int) s.pop());
+        s.push(1);
+        s.push(2);
+        s.push(3);
+        assertEquals(3, (int) s.pop());
+        assertEquals(2, (int) s.pop());
+        s.push(4);
+        assertEquals(4, (int) s.pop());
+        assertEquals(1, (int) s.pop());
+    }
+
     @Test
     public void testStackMultipleInts() {
-        ArrayList<Integer> expected = new ArrayList<>();
-        for (int i = 2; i < 10; i = i + 2) {
-            expected.add(i);
-        }
+        Stack<Integer> s = new Stack<>();
+        s.push(1);
+        s.push(2);
+        s.push(3);
 
-        Stack s = new Stack();
-        for (int e : expected) {
-            s.push(e);
-        }
+        assertEquals(3, (int) s.pop());
+        assertEquals(2, (int) s.pop());
+        assertEquals(1, (int) s.pop());
+    }
+
+    @Test
+    public void testGrowth() {
+        Stack<Integer> s = new Stack<>(1);
+        int startingCapacity = s.capacity();
```

Comment:
You know what the capacity should be at this point, 1, so I'd just assertEquals(1, s.capacity())

Reply:
Since the test's stated goal is the internal growth function, what's the idea on assertions that might be orthogonal? Should a sanity check for initial capacity be in a separate test? Should a dependant states be tested along with the ostensible test? Both?